public/app/core/logs_model.ts

```diff
@@ -311,6 +314,8 @@ export function logSeriesToLogsModel(logSeries: DataFrame[]): LogsModel | undefi
     for (let j = 0; j < series.length; j++) {
       const ts = timeField.values.get(j);
       const time = dateTime(ts);
+      const tsNs = isLoki ? timeNanosecondField.values.get(j) : undefined;
```

Comment:
Isn't `timeNanosecondField` going to be always there as we have created it couple lines above? And therefoore it is going to either give us undefined or value?

Reply:
`timeNanosecondField` can either have values or be undefined. Would personally use an optional chaining operator there but got asked to write it in a longer form as it's easier to read.

Reply:
But shouldn't it be then `const tsNs = timeNanosecondField ? timeNanosecondField.values.get(j) : undefined;` Or in my opinion, this is mor readable `const tsNs = timeNanosecondField?.values.get(j);`

Reply:
Yep, can be done that way, but there's one downside of this choice: we can never be sure that other datasources are going to have a timestamp-format data under this field. I'll write some additional logic to check that.